clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java

```diff
@@ -630,6 +665,21 @@ public CompletableFuture<Void> acknowledgeOnClose(final Map<TopicIdPartition, No
         return closeFuture;
     }
 
+    private boolean isNodeLeader(int nodeId, TopicIdPartition topicIdPartition) {
+        Optional<Node> leaderNode = metadata.currentLeader(topicIdPartition.topicPartition()).leader;
+        if (leaderNode.isPresent()) {
+            if (leaderNode.get().id() != nodeId) {
+                log.debug("Node {} is no longer the leader for topic ID partition {}, failing acknowledgements", nodeId, topicIdPartition);
```

Comment:
nit: In this file, the log lines just use `for partition {}` and never `for topic ID partition {}`.